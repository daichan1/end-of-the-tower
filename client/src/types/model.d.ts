export type EnemyType = {
  id: number
  name: string
  imageUrl: string
  hp: number
  attack: number
  defense: number
}

export type PlayerType = {
  name: string
  imageUrl: string
  hp: number
  attack: number
  defense: number
  energy: number
}

export type CardType = {
  name: string
  description: string
  imageUrl: string
  cost: number
  cardType: string
  attack: number
  defense: number
}
