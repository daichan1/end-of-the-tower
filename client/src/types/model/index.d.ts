export type EnemyType = {
  id: number
  name: string
  imageUrl: string
  hp: number
  maxHp: number
  attack: number
  defense: number
  damage: number
  isDamaged: boolean
}

export type PlayerType = {
  name: string
  imageUrl: string
  hp: number
  maxHp: number
  attack: number
  defense: number
  energy: number
  stage: number
  deck: CardType[]
  nameplate: CardType[]
  cemetery: CardType[]
}

export type CardBaseType = {
  name: string
  description: string
  imageUrl: string
  cost: number
  cardType: string
  attack: number
  defense: number
  actionName: string
  executionCount: number
  effectType: string
}

export type CardType = {
  id: number
  name: string
  description: string
  imageUrl: string
  cost: number
  cardType: string
  attack: number
  defense: number
  actionName: string
  executionCount: number
  effectType: string
}
