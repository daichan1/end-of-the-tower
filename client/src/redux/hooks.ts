import { useSelector, useDispatch, TypedUseSelectorHook } from 'react-redux'
import type { RootStore, AppDispatch } from './store'

// 型付きのuseSelectorを事前に定義
export const useAppSelector: TypedUseSelectorHook<RootStore> = useSelector
// 型付きのuseDispatchを事前に定義
export const useAppDispatch = () => useDispatch<AppDispatch>()
