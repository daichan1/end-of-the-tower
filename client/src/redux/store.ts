import { configureStore } from '@reduxjs/toolkit'
import playerReducer from './slice/playerSlice'
import enemiesReducer from './slice/enemiesSlice'
import cardsReducer from './slice/cardsSlice'
import fightEnemiesReducer from './slice/fightEnemiesSlice'
import gameTitleReducer from './slice/gameTitleSlice'
import rootSelectReducer from './slice/rootSelectSlice'
import battleReducer from './slice/battleSlice'
import turnReducer from './slice/turnSlice'

export const store = configureStore({
  reducer: {
    player: playerReducer,
    enemies: enemiesReducer,
    cards: cardsReducer,
    fightEnemies: fightEnemiesReducer,
    gameTitle: gameTitleReducer,
    rootSelect: rootSelectReducer,
    battle: battleReducer,
    turn: turnReducer
  }
})

export type RootStore = ReturnType<typeof store.getState>
export type AppDispatch = typeof store.dispatch
