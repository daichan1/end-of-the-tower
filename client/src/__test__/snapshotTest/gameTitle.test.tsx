import renderer from 'react-test-renderer'
import { useAppSelector, useAppDispatch } from '../../redux/hooks'
import { CardType } from '../../types/model'
import GameTitle from '../../components/gameTitle'

jest.mock('../../redux/hooks')
const useAppSelectorMock = useAppSelector as jest.Mock
const useAppDispatchMock = useAppDispatch as jest.Mock

describe("GameTitle Component Snapshot", () => {
  const cards: CardType[] = []
  let gameTitle = false
  beforeEach(() => {
    useAppSelectorMock.mockReturnValueOnce(cards)
    useAppDispatchMock.mockReturnValue(jest.fn())
  })
  afterEach(() => {
    jest.resetAllMocks()
  })

  it("render game title component", () => {
    useAppSelectorMock.mockReturnValueOnce(gameTitle)
    const component = renderer.create(<GameTitle />).toJSON()

    expect(component).toMatchSnapshot()
  })

  it("not render game title component", () => {
    gameTitle = true
    useAppSelectorMock.mockReturnValueOnce(gameTitle)
    const component = renderer.create(<GameTitle />).toJSON()

    expect(component).toMatchSnapshot()
  })
})
