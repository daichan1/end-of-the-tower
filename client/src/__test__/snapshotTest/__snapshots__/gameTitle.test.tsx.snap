// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameTitle Component Snapshot not render game title component 1`] = `
<div
  className="game-title"
  style={
    Object {
      "display": "none",
    }
  }
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-1ewlecp-MuiGrid-root"
    style={
      Object {
        "height": 600,
      }
    }
  >
    <h1
      className="title"
    >
      End of the Tower
    </h1>
    <button
      className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      ゲームスタート
      <span
        className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;

exports[`GameTitle Component Snapshot render game title component 1`] = `
<div
  className="game-title"
  style={
    Object {
      "display": "",
    }
  }
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column css-1ewlecp-MuiGrid-root"
    style={
      Object {
        "height": 600,
      }
    }
  >
    <h1
      className="title"
    >
      End of the Tower
    </h1>
    <button
      className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      ゲームスタート
    </button>
  </div>
</div>
`;
